idb/postgres/postgres.go

```diff
@@ -242,60 +241,87 @@ func (db *IndexerDb) AddBlock(block *bookkeeping.Block) error {
 			}
 		}
 
+		return nil
+	}
+	// `ff` takes ownership of `tx`. We split functionality in two functions,
+	// `f` and `ff`, so that we can easily free the database resources once and only once
+	// using defer before committing the database transaction.
+	ff := func(tx pgx.Tx) error {
+		err := f(tx)
+		if err != nil {
+			tx.Rollback(context.Background())
```

Comment:
Can this go in `TxWithRetry`? It would be easy to forget this in the future.

Reply:
Good idea!